maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveJsonSchema.java

```diff
@@ -89,17 +91,27 @@ protected ObjectNode mergeMultipleDataTypes(ObjectNode mergedArray,
     return mergedArray;
   }
 
-  public ObjectNode getSchemaForMultipleMessages(List<String> messages)
+  /**
+   * Treated same as array of records, the items derived is returned as schema
+   * Exactly one schema is returned
+   */
+  @Override
+  public ObjectNode getSchemaForMultipleMessages(List<JsonNode> messages)
       throws JsonProcessingException {
-    // Get schema for multiple messages. Exactly one schema is returned
-    // Treated same as array of records, the items derived is returned
-    List<JsonNode> messageObjects = new ArrayList<>();
-    for (String message : messages) {
-      messageObjects.add(mapper.readTree(message));
-    }
-    JsonNode schema = getSchemaForArray(messageObjects, "").get("items");
+    JsonNode schema = getSchemaForArray(messages, "").get("items");
+    convertToFormat(schema, "");
```

Comment:
Is return value being ignored on purpose? The name of this function implies that we want to convert to some format, so the converted output should be used in the calling function.

Reply:
I'll make changes to use the output. Ideally, we would like to use the JsonSchema generated to be returned. I'll make changes to have that